### PR TITLE
Q-YO Blaster

### PR DESCRIPTION
**Pull request by tabreturn**
# New Port for Q-YO Blaster

## Game Information
- **Title**: Q-YO Blaster
- **URL**: https://teamblackhatrobot.itch.io/q-yo-blastergame
### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
Test all major resolutions:
- [ ] 480x320
- [x] 640x480
- [x] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/qyoblaster/Q-YO Blaster.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+# Variables
+GAMEDIR="/$directory/ports/qyoblaster"
+
+# CD and set permissions
+cd $GAMEDIR
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+$ESUDO chmod +x -R $GAMEDIR/*
```
Comment (cdeletre) [inline ports/qyoblaster/Q-YO Blaster.sh:24]: When possible use a fine permission set
```suggestion
$ESUDO chmod +x $GAMEDIR/gmloadernext
$ESUDO chmod +x $GAMEDIR/tools/SDL_swap_gpbuttons.py
$ESUDO chmod +x $GAMEDIR/tools/gmKtool.py
$ESUDO chmod +x $GAMEDIR/tools/oggdec
$ESUDO chmod +x $GAMEDIR/tools/oggenc
$ESUDO chmod +x $GAMEDIR/tools/patchscript
$ESUDO chmod +x $GAMEDIR/tools/xdelta3

File: ports/qyoblaster/Q-YO Blaster.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        echo "This port requires the latest version of PortMaster." > $CUR_TTY
+    fi
+else
+    echo "Patching process already completed. Skipping."
+fi
+
+# Display loading splash
+if [ -f "$GAMEDIR/gamedata/game.droid" ]; then
+[ "$CFW_NAME" == "muOS" ] && $ESUDO ./tools/splash "splash.png" 1 # muOS only workaround
+    $ESUDO ./tools/splash "splash.png" 2000
```
Comment (cdeletre) [inline ports/qyoblaster/Q-YO Blaster.sh:54]: `./tools/splash` is missing

File: ports/qyoblaster/Q-YO Blaster.sh (hunk 1 above)
Comment (cdeletre) [reply ports/qyoblaster/Q-YO Blaster.sh:24]: you may need to add
`$ESUDO chmod +x $GAMEDIR/tools/splash`

File: ports/qyoblaster/Q-YO Blaster.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+dos2unix "$GAMEDIR/tools/Klib/GMblob.py"
+dos2unix "$GAMEDIR/tools/patchscript"
+
+# Check if patchlog.txt to skip patching
+if [ ! -f patchlog.txt ]; then
+    if [ -f "$controlfolder/utils/patcher.txt" ]; then
+        source "$controlfolder/utils/patcher.txt"
+        $ESUDO kill -9 $(pidof gptokeyb)
+    else
+        echo "This port requires the latest version of PortMaster." > $CUR_TTY
```
Comment (cdeletre) [inline ports/qyoblaster/Q-YO Blaster.sh:45]: We now have this magic function https://github.com/PortsMaster/PortMaster-GUI/blob/fe7756b9c7a523af54b6ef44f1d3c536512091a2/PortMaster/funcs.txt#L108

```suggestion
        pm_message "This port requires the latest version of PortMaster."
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

File: ports/qyoblaster/Q-YO Blaster.sh (hunk 2 above)
Comment (tabreturn) [reply ports/qyoblaster/Q-YO Blaster.sh:61]: Well spotted. Thx, I've added it.

**Review by Cebion — APPROVED** [review]

**Pull request merged**